tests/test_client.py

```diff
@@ -265,3 +268,49 @@ async def test_e2e_consume_multiple_topics():
         assert topic_2.total_events == total_events
 
         assert TopicManager.all_messages_consumed()
+
+
+@pytest.mark.asyncio
+async def test_streams_consume_events_with_initial_offsets(stream_engine: StreamEngine):
+    client = TestStreamClient(stream_engine)
+    event1 = b'{"message": "Hello world1!"}'
+    event2 = b'{"message": "Hello world2!"}'
+    process = Mock()
+
+    async with client:
+        await client.send(topic, value=event1, partition=0)
+        await client.send(topic, value=event1, partition=0)
+        await client.send(topic, value=event2, partition=1)
+
+        async def func_stream(consumer: Stream):
+            async for cr in consumer:
+                process(cr.value)
+
+        stream: Stream = Stream(
+            topics=topic,
+            consumer_class=TestConsumer,
+            name="my-stream",
+            func=func_stream,
+            initial_offsets=[
+                TopicPartitionOffset(topic=topic, partition=0, offset=1),
+                TopicPartitionOffset(topic=topic, partition=1, offset=1),
+            ],
+        )
+        stream_engine.add_stream(stream)
+        await stream.start()
+
+        assert stream.consumer.assignment() == [tp0, tp1, tp2]
+
+        assert stream.consumer.last_stable_offset(tp0) == 2
+        assert stream.consumer.highwater(tp0) == 2
+        assert await stream.consumer.position(tp0) == 2
+
+        assert stream.consumer.last_stable_offset(tp1) == 1
+        assert stream.consumer.highwater(tp1) == 1
+        assert await stream.consumer.position(tp1) == 1
+
+        assert stream.consumer.last_stable_offset(tp2) == 0
+        assert stream.consumer.highwater(tp2) == 0
+        assert await stream.consumer.position(tp2) == 0
+
+    process.assert_called()
```

Comment:
I think this test needs slight editing.

Given 2 partitions, and event 1,2,3,4, I would send 1 and 2 to partition 0 and 2 and 3 to partition 2. Then you can change the assert at the end to say assert_called_with( event 2 and 4), and assert not called with 1 and 3.

Reply:
This is not how the mocks works, you can not do that. You can only assert to the latest called if you use `assert_called_with`, in this case you can do `process.assert_called_with(event2)`. If you do `process.assert_called_with(event1)` it will fail. What we could do is `process.assert_has_calls(...)`

Reply:
You can not use any order with `assert_has_calls`. It has to be the same order that the `mock` was called. Anyway, the `goal` is to test the `seek` and which will be thenext `offset` after a `seek` operation regardless the event order or with which events that mock was called. If the next offset is the expected one, let's say `assert stream.consumer.last_stable_offset(tp0) == 2` it means that the events have been consumed which it makes the mock unnecessary.

Reply:
fair point about the point of the test. Consumption is not necessary. Checking the next offset is enough. 

assert_has_calls has an any_order param that defauls to False.